rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -442,15 +467,21 @@ def _deserialize_adex_event(
     ) -> AdexEvent:
         """Deserialize the common event attributes.
 
-        It may raise KeyError.
         Id for unbond and unbond request events is 'tx_hash:address'.
+        May raise:
+        - KeyError
+        - DeserializationError: via `deserialize_timestamp()`.
         """
         identity_address = to_checksum_address(raw_event['owner'])
+        tx_hash = raw_event['id'].split(':')[0]
```

Comment:
this can raise ValueError ... what did I tell you in the last review? Are you even reading my reviews?

You are not even mentioning it in the docstring of the function and you are not even handling it in all the places where `deserialize_adex_event()` is called. You are not handling it:

- [here](https://github.com/vnavascues/rotki/blob/c98babbe3e0942d08d6e5669171b0a37de6a33e6/rotkehlchen/chain/ethereum/adex/adex.py#L502)
- [here](https://github.com/vnavascues/rotki/blob/c98babbe3e0942d08d6e5669171b0a37de6a33e6/rotkehlchen/chain/ethereum/adex/adex.py#L537)

Additionally what did I write about `KeyError` and `ValueError`? The error message **makes no sense**

Do you know what `str(e)` is for a key error? It's the freaking missing  key strung. That's all.

So when you try to handle it [here](https://github.com/vnavascues/rotki/blob/c98babbe3e0942d08d6e5669171b0a37de6a33e6/rotkehlchen/chain/ethereum/adex/adex.py#L385):

The user will see: Failed to deserialize an AdEx bond event. Check logs for more details. error='timestamp'. That is bad. That is what I said in the last review. Please don't make me repeat myself in the reviews of the same day. I need to be able to trust you. Can't spend so much time reviewing your code.

You can still leave it as you have handling the KeyError out there. That's fine and indeed is much better to avoid code duplicate. But make the message make sense. Like we do [here](https://github.com/rotki/rotki/blob/70508f99f890bcbd520f1efe7776194d6a5e5e06/rotkehlchen/exchanges/poloniex.py#L185-L198) in poloniex.


Same comments apply about letting the ValueError propagate up.

Reply:
1. ValueError would be raised by `to_checksum_address`. Fair enough, I will handle it here and in the callers. I will also add it in the docstring and add a specific message.
2. `raw_event['id'].split(':')[0]` can only raise AttributeError (if 'id' is not a string), as I'm accessing to index 0. I understand though your concerns on line 420 where actually there are values to unpack (ValueError is captured there, but with a vague message you dislike). Do you want me to specifically handle any AttributeError when splitting? I haven't seen anything that defensive on any subgraph/exchange entry deserialization.
3. The users won't see `Failed to deserialize an AdEx bond event. Check logs for more details. error='timestamp'.` but this (as Poloniex does):
![image](https://user-images.githubusercontent.com/14769962/107278023-27f00700-6a4d-11eb-9dfd-6af960053d3f.png)

I decided to omit to the user the event data as it contains multiple fields with very long strings (e.g. pool id, channel id), there is no point for them to see that.

I can extend the error message with `Missing key entry for < key >` for error logs due to KeyError, as Poloniex does. I understand it will help us. But what really matters from these error logs is the `raw_event`.

Reply:
> ValueError would be raised by to_checksum_address. Fair enough, I will handle it here and in the callers. I will also add it in the docstring and add a specific message.

Capture and handle it properly

> raw_event['id'].split(':')[0] can only raise AttributeError (if 'id' is not a string), as I'm accessing to index 0. I understand though your concerns on line 420 where actually there are values to unpack (ValueError is captured there, but with a vague message you dislike). Do you want me to specifically handle any AttributeError when splitting? I haven't seen anything that defensive on any subgraph/exchange entry deserialization.

Indeed. I was looking at line 420. But what you mentioned is even worse. If the subgraph returns something without the `:` you won't notice it and you will have a mistaken transaction hash in the event.

We are being defensive when it comes to external data. Always. What do you think all these `deserialize_xxxx` functions do? Sanitizing external user input. At this point since you use this `:` splitting in a few places in the module I would say to make a local `deserialize_xxx()` function which also checks that what is passed is a string.

> The users won't see Failed to deserialize an AdEx bond event

I meant the users who look at the logs. And also us who will end up looking at the log. Please fix this. I think I made myself quite very clear and even pointed to another part in the code with what you need to do.

> I can extend the error message with Missing key entry for < key > for error logs due to KeyError, as Poloniex does. I understand it will help us.

Yes this is what I am asking.